src/crate/client/cursor.py

```diff
@@ -210,3 +218,23 @@ def duration(self):
                 "duration" not in self._result:
             return -1
         return self._result.get("duration", 0)
+
+    def convert_rows(self):
+        """
+        Iterate rows, apply type converters, and generate converted rows.
+        """
+        assert "col_types" in self._result, \
+            "Unable to apply type conversion without `col_types` information"
+        type_id_list = self._result["col_types"]
+        for row in self._result["rows"]:
+            yield [
+                self._converter.convert(type_id, value)
+                for type_id, value in zip(type_id_list, row)
+            ]
```

Comment:
Hi @matriv,

at https://github.com/crate/crate-python/pull/437#discussion_r924614148, you asked:

> Can we do [this] more elegant?

Maybe this implementation fits your eyes better.

With kind regards,
Andreas.